packages/block-editor/src/store/reducer.js

```diff
@@ -1727,6 +1727,31 @@ export function highlightedBlock( state, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning the block insertion event list state.
+ *
+ * @param {Object}  state  Current state.
+ * @param {Object}  action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function lastBlockInserted( state = {}, action ) {
+	switch ( action.type ) {
+		case 'INSERT_BLOCKS':
+			if ( ! action.updateSelection || ! action.blocks.length ) {
```

Comment:
After a second thought, I think that we probably shouldn't include the `! action.updateSelection` condition here. The last block inserted should be updated independently on if the block gets selected upon insertion, right?

Originally I thought that this was necessary because we only wanted to show the media picker when the block is selected, but I realised that we're already checking that in the components ([example](https://github.com/WordPress/gutenberg/pull/29546/files#diff-7be1df22c250a7378a0796b6c68da7fc1178a98d9e88ab9f59868a6d0b8c467fR405)), what do you think?

Reply:
> After a second thought, I think that we probably shouldn't include the ! action.updateSelection condition here. The last block inserted should be updated independently on if the block gets selected upon insertion, right?

when you say "updated independently on if the block gets selected upon insertion" what exactly do you mean? I want to ensure that I understand you clearly :) 

> Originally I thought that this was necessary because we only wanted to show the media picker when the block is selected, but I realised that we're already checking that in the components (example), what do you think?

Makes sense! I checked the linked example and it went to the `withSelect` in the Gallery component, so I am assuming that you are referring to the `isSelected` usage there. If that's the case, yes, the `isSelected` behavior  was added before the `redux` approach was introduced, so if it's better to keep the selection logic within the reducer, I am not against that, especially if it will be consistent with other behavior in the reducer.

Reply:
>when you say "updated independently on if the block gets selected upon insertion" what exactly do you mean? I want to ensure that I understand you clearly :)

Oh sorry if I wasn't too clear, what I meant is that if this slice of the state contains the last block inserted, it makes sense that it gets updated for any block that is inserted, including the ones with the param `updateSelection` as false. From a consistence POV, I would expect that if I insert a block even if it doesn't get automatically selected (via the `updateSelection` action param), it would be set as the last block inserted.

>Makes sense! I checked the linked example and it went to the withSelect in the Gallery component, so I am assuming that you are referring to the isSelected usage there. If that's the case, yes, the isSelected behavior was added before the redux approach was introduced, so if it's better to keep the selection logic within the reducer, I am not against that, especially if it will be consistent with other behavior in the reducer.

Yeah exactly, I was referring to the `isSelected` usage when calculating the value to pass to `autoOpenMediaUpload` prop.

In my opinion I think it would make more sense to keep the `isSelected` usage as we have currently in the components and update the `lastBlockInserted` reducer.

>Another question, I wanted to add to the discussion. In which cases would the block be inserted and it wouldn't be a selection. I removed isSelected and updateSelection from the code in my local environs and the functionality still worked. I will look more into this tomorrow, but I just wanted to share my thoughts here, so you can add any ideas you may have.

I think it's a uncommon case but looks like there's at least one place where it's being used in the Jetpack project:
https://github.com/Automattic/jetpack/blob/5b637a82a9836213f066809cab550ce3590921cb/projects/plugins/jetpack/extensions/blocks/anchor-fm/editor.js#L48

Reply:
> Oh sorry if I wasn't too clear, what I meant is that if this slice of the state contains the last block inserted, it makes sense that it gets updated for any block that is inserted, including the ones with the param updateSelection as false. From a consistence POV, I would expect that if I insert a block even if it doesn't get automatically selected (via the updateSelection action param), it would be set as the last block inserted.

Ah I get what you are saying now. Thanks for the clarification 🙇🏾 

> In my opinion I think it would make more sense to keep the isSelected usage as we have currently in the components and update the lastBlockInserted reducer.

I agree with this as well after gaining a deeper understanding due to your clarifications above. I will make the relevant changes to support this. 

> I think it's a uncommon case but looks like there's at least one place where it's being used in the Jetpack project:
https://github.com/Automattic/jetpack/blob/5b637a82a9836213f066809cab550ce3590921cb/projects/plugins/jetpack/extensions/blocks/anchor-fm/editor.js#L48

Good find 😉 Thanks for checking Jetpack. If I had a sleuthing badge 🔍  I would definitely give you one 😄

Reply:
>I agree with this as well after gaining a deeper understanding due to your clarifications above. I will make the relevant changes to support this.

Nice! Thanks for making the changes, I'll be watching them so we can have this PR ready as soon as possible.

>Good find 😉 Thanks for checking Jetpack. If I had a sleuthing badge 🔍 I would definitely give you one 😄
Thank you very much! 🕵️😄